### PR TITLE
base16384: bump to version 2.2.4

### PR DESCRIPTION
**Pull request by fumiama**
Maintainer: me
Compile tested: x86/64, Generic PC, OpenWrt 22.03.2 r19803-9a599fee93
Run tested: x86/64, Generic PC, OpenWrt 22.03.2 r19803-9a599fee93

Signed-off-by: 源 文雨 <fumiama@foxmail.com>

### CODE REVIEW

File: utils/base16384/test.sh [hunk 1]
```diff
@@ -3,9 +3,9 @@
 case "$1" in
 	"base16384")
         i=1
-        while [ $i -le 100 ]; do
-            s="$(head /dev/urandom | head -c $i)"
-            if [ "$(echo $s)" != "$(echo $s | base16384 -e - - | base16384 -d - -)" ]; then
+        while [ $i -le 256 ]; do
+            s="$(head /dev/urandom | head -c $i | base64 -w 0)"
```
Comment (1715173329) [inline utils/base16384/test.sh:7]: Command base64 is not included in openwrt by default, you may probably need to install `coreutils-base64` firstly. E.g. insert `opkg install `

Comment (fumiama) [reply utils/base16384/test.sh:8]: Thank you for your instruction 🤝. Fixed it.

**Comment by 1715173329** [comment]
mips runtime test is failing, is it expected?

**Comment by fumiama** [comment]
> mips runtime test is failing, is it expected?

It is not what I had expected and it seems to happen after I added `opkg install`.

When I open this PR with only a version changing, I did get this mips arch error. So I added base64 to test whether it is caused by the binary-echo issue, then all test passed so a think it has been fixed. However, after I added `opkg install`, it failed again.

Here are two tests in my repo.

1. https://github.com/fumiama/packages/pull/2 mips failed
2. https://github.com/fumiama/packages/pull/1 all tests passed

**Comment by 1715173329** [comment]
> 2. [test: without opkg install fumiama/packages#1](https://github.com/fumiama/packages/pull/1) all tests passed

It's not really "passed" as in logs it prints "`/ci/utils/base16384/test.sh: line 7: base64: not found`"

**Comment by fumiama** [comment]
Sorry, I read the log again and found that the mips test passed because base64 not found. So it seems to be the program's own issue. I will do more test and try to fix it soon.

**Comment by fumiama** [comment]
Thanks for your instruction, I have detected a bug and released a new version 2.2.4. Hope no problem this time 😂.

**Comment by 1715173329** [comment]
Good job, thanks!

**Comment by 1715173329** [comment]
would you mind me cherry-picking (backporting) this commit to the 23.05 branch?

**Comment by fumiama** [comment]
> would you mind me cherry-picking (backporting) this commit to the 23.05 branch?

No, I'd like it.

**Comment by 1715173329** [comment]
done in fb4b7a4aaaf668d958754aab4d052628d309250e ;)

**Pull request merged**